packages/react-data-grid/src/common/types.ts

```diff
@@ -29,3 +29,23 @@ export interface CellMetaData {
   onCellExpand(): void;
   getCellActions?(): void;
 }
+
+export interface Position {
+  idx: number;
+  rowIdx: number;
+}
+
+export interface Range {
+  topLeft: Position;
+  bottomRight: Position;
+}
+
+export interface Dimension {
+  width: number;
+  height: number;
+  top: number;
+  left: number;
+  zIndex: number;
+}
+
+export type RowGetter = (rowIdx: number) => unknown;
```

Comment:
Should we consider a generic type?
```
export type RowGetter = <R = any>(rowIdx: number) => R;
```

Reply:
I'm not sure yet.

I want to avoid `<R = any>` for now, as it's likely users won't set the generic type and `any` isn't useful.

`<R>` alone defaults to an empty object `{}` if the generic isn't defined.
![image](https://user-images.githubusercontent.com/567105/56024394-d7e87200-5d07-11e9-9cc0-81a602e10aaf.png)

I think we should just revisit this later.